### PR TITLE
Add a SLIP client for OSC over serial ports.

### PR DESCRIPTION
**Pull request by MartijnBraam**
The OSC specification lets you send messages over any framed data channel and using the framing part of SLIP seems to be the most popular solution for devices allowing OSC control over USB.

### CODE REVIEW

File: pythonosc/slip_client.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    def __init__(self, device, baudrate=9600, **kwargs):
+        self.ser = serial.Serial(device, baudrate=baudrate, **kwargs)
+
+    def send(self, content: Union[OscMessage, OscBundle]) -> None:
+        encoded = self.END
+        encoded += content.dgram.replace(self.ESC, self.ESC + self.ESC_ESC).replace(self.END, self.ESC + self.ESC_END)
+        encoded += self.END
+        self.ser.write(encoded)
+
+    def receive(self):
```
Comment (attwad) [inline pythonosc/slip_client.py:24]: can you please add some docs on what to expect as return type please (OscMessage)

File: pythonosc/slip_client.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        buffer = b''
+        while True:
+            c = self.ser.read(1)
+            if c is None:
+                return None
```
Comment (attwad) [inline pythonosc/slip_client.py:29]: just `return` please, no need for None

File: pythonosc/slip_client.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+            if c == self.END:
+                if len(buffer):
+                    break
+                continue
+
+            if c == self.ESC:
+                c = self.ser.read(1)
+                if c is None:
+                    return None
```
Comment (attwad) [inline pythonosc/slip_client.py:39]: same here

**Review by attwad — COMMENTED** [review]
Seems like tests are unhappy with the serial import, I don't think this is part of the standard library right?  You could add the necessary dependency to https://github.com/attwad/python-osc/blob/master/.github/workflows/python-test.yml to make them pass.

**Comment by MartijnBraam** [comment]
Fixed up the receive method and added pyserial as dependency.

I wonder if it's better to return an exception instead of None when the serial port dissapears. The upside of None is that it follows the socket API a bit more closely.

**Comment by attwad** [comment]
> Fixed up the receive method and added pyserial as dependency.
> 
> I wonder if it's better to return an exception instead of None when the serial port dissapears. The upside of None is that it follows the socket API a bit more closely.

Fine as is imo, thanks.

Tests are still failing but I don't think you want to write a whole set of stub types for the pyserial package... fine to suppress with # type: ignore imo. Cf. https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports

**Comment by MartijnBraam** [comment]
Seems like there are also errors in files I haven't touched and some tests complaining about the removed None (on 3.7)
